polaris/ocean/convergence/spherical/analysis.py

```diff
@@ -0,0 +1,409 @@
+import datetime
+import os
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+from polaris import Step
+from polaris.ocean.resolution import resolution_to_subdir
+
+
+class SphericalConvergenceAnalysis(Step):
+    """
+    A step for analyzing the output from the geostrophic convergence test
+
+    Attributes
+    ----------
+    resolutions : list of float
+        The resolutions of the meshes that have been run
+
+    icosahedral : bool
+        Whether to use icosahedral, as opposed to less regular, JIGSAW
+        meshes
+
+    dependencies_dict : dict of dict of polaris.Steps
+        The dependencies of this step must be given as separate keys in the
+        dict:
+            mesh : dict of polaris.Steps
+                Keys of the dict correspond to `resolutions`
+                Values of the dict are polaris.Steps, which must have the
+                attribute `path`, the path to `base_mesh.nc` of that
+                resolution
+            init : dict of polaris.Steps
+                Keys of the dict correspond to `resolutions`
+                Values of the dict are polaris.Steps, which must have the
+                attribute `path`, the path to `initial_state.nc` of that
+                resolution
+            forward : dict of polaris.Steps
+                Keys of the dict correspond to `resolutions`
+                Values of the dict are polaris.Steps, which must have the
+                attribute `path`, the path to `forward.nc` of that
+                resolution
```

Comment:
Sphinx isn't happy with this syntax.  These docstrings have a pretty specific format that they support.  I'm seeing:
```
/home/xylar/code/e3sm/polaris/ocn-shared-convergence-analysis-step/polaris/ocean/convergence/spherical/analysis.py:docstring of polaris.ocean.convergence.spherical.analysis.SphericalConvergenceAnalysis:12: ERROR: Unexpected indentation.
/home/xylar/code/e3sm/polaris/ocn-shared-convergence-analysis-step/polaris/ocean/convergence/spherical/analysis.py:docstring of polaris.ocean.convergence.spherical.analysis.SphericalConvergenceAnalysis:31: ERROR: Unexpected indentation.
/home/xylar/code/e3sm/polaris/ocn-shared-convergence-analysis-step/polaris/ocean/convergence/spherical/analysis.py:docstring of polaris.ocean.convergence.spherical.analysis.SphericalConvergenceAnalysis.__init__:14: ERROR: Unexpected indentation.
/home/xylar/code/e3sm/polaris/ocn-shared-convergence-analysis-step/polaris/ocean/convergence/spherical/analysis.py:docstring of polaris.ocean.convergence.spherical.analysis.SphericalConvergenceAnalysis.__init__:32: ERROR: Unexpected indentation.
```

Reply:
It seems like we just need a blank line.  The warning goes away when we have that.